addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/factory/MySensorsCacheFactory.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mysensors.factory;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+/**
+ * Cache may be used to store information across restart of MySensors connection.
+ *
+ * @author Andrea Cioni
+ *
+ */
+public class MySensorsCacheFactory {
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    private String CACHE_BASE_PATH;
+    private static final String CACHE_FILE_SUFFIX = ".cached";
+
+    public static final String GIVEN_IDS_CACHE_FILE = "given_ids";
+
+    private final Gson gson = new Gson();
+
+    public MySensorsCacheFactory(String userDataFolder) {
+        CACHE_BASE_PATH = userDataFolder + "/mysensors/cache";
```

Comment:
This is a field now, please follow coding conventions and make the name small and camelcase